lib/puppet/metatype/manager.rb

```diff
@@ -143,29 +143,29 @@ def rmtype(name)
   #
   def type(name)
     # Avoid loading if name obviously is not a type name
-    if name.to_s.include?(':')
-      return nil
-    end
+    return nil if name.to_s.include? ':'
 
     @types ||= {}
 
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
This part (and the same change a few lines below) seems safe to me. Is there any downside? 

My only thought is whether it could be that it's an optimization in the one case, where the `@types` hash doesn't already contain `name`, but a pessimization where it does. That would take a little benchmarking of ruby.